### PR TITLE
Add the missing pieces for NetworkService support

### PR DESCRIPTION
**Pull request by tangenti**

### CODE REVIEW

File: pkg/cloud/gen/main.go [hunk 1]
```diff
@@ -1271,12 +1272,14 @@ const location = "location"
...
-		"BetaComputePackage":  betaComputePackage,
-		"GaComputePackage":    gaComputePackage,
+		"Year":                       fmt.Sprintf("%v", time.Now().Year()),
+		"FilterPackage":              filterPackage,
+		"MetaPackage":                metaPackage,
+		"AlphaComputePackage":        alphaComputePackage,
+		"BetaComputePackage":         betaComputePackage,
+		"GaComputePackage":           gaComputePackage,
+		"GANetworkservicesPackage":   gaNetworkServicesPackage,
+		"BetaNetworkservicesPackage": betaNetworkServicesPackage,
```
Comment (robscott) [inline pkg/cloud/gen/main.go]: Tiny nits: "Ga" vs "GA" is inconsistent here. Ordering also doesn't match (alpha > beta > ga followed by ga > beta)

File: pkg/cloud/gen/main.go [hunk 2]
```diff
@@ -886,17 +881,20 @@ func (g *{{.GCEWrapType}}) Insert(ctx context.Context, key *meta.Key, obj *{{.FQ
...
+		klog.V(4).Infof("{{.GCPWrapType}}.Insert(%v, %v, ...): RateLimiter error: %v", ctx, key, err)
 		return err
 	}
 	obj.Name = key.Name
 
 {{- if .IsNetworkServices}}
 	parent := fmt.Sprintf("projects/%s/locations/global", projectID)
 	call := g.s.{{.GroupVersionTitle}}.{{.Service}}.Create(parent, obj)
+	{{- if eq .Object "TcpRoute"}}
+	  call.TcpRouteId(obj.Name)
```
Comment (robscott) [inline pkg/cloud/gen/main.go]: Assuming this applies more broadly - do you know which other APIs would require something like this? If we leave this hardcoded in we may end up getting unnecessarily stuck debugging later whenever we add more resources like HTTPRoute.

File: pkg/cloud/gen/main.go [hunk 3]
```diff
@@ -886,17 +881,20 @@ func (g *{{.GCEWrapType}}) Insert(ctx context.Context, key *meta.Key, obj *{{.FQ
 		Service: "{{.Service}}",
 	}
 
-	klog.V(5).Infof("{{.GCEWrapType}}.Insert(%v, %v, ...): projectID = %v, ck = %+v", ctx, key, projectID, ck)
+	klog.V(5).Infof("{{.GCPWrapType}}.Insert(%v, %v, ...): projectID = %v, ck = %+v", ctx, key, projectID, ck)
```
Comment (robscott) [inline pkg/cloud/gen/main.go:887]: I think this is intentionally out of scope for this PR, but we may want to update the logs for NetworkServices API calls to be more reflective of the calls we're actually making.

**Review by robscott — COMMENTED** [review]
Thanks @tangenti! A couple tiny nits but otherwise LGTM.

File: pkg/cloud/gen/main.go (hunk 1 above)
Comment (tangenti) [reply pkg/cloud/gen/main.go]: Done

File: pkg/cloud/gen/main.go (hunk 2 above)
Comment (tangenti) [reply pkg/cloud/gen/main.go]: It seems all the routes require the ID. I've made the change to append the ID for all routes.

File: pkg/cloud/gen/main.go (hunk 3 above)
Comment (tangenti) [reply pkg/cloud/gen/main.go:887]: Corrected it to `Create` for TD resources.
Would you like to revert the change for `GCEWrapType` to `GCPWrapType`?

Comment (robscott) [reply pkg/cloud/gen/main.go:887]: Thanks! I think the `GCEWrapType` to `GCPWrapType` rename makes sense, good to leave that in place.

**Review by robscott — APPROVED** [review]
Thanks @tangenti!

**Pull request merged**